sdk/core/azure-core/src/main/java/com/azure/core/http/ProxyOptions.java

```diff
@@ -68,6 +115,125 @@ public String getPassword() {
         return this.password;
     }
 
+    /**
+     * @return the hosts that bypass the proxy.
+     */
+    public String getNonProxyHosts() {
+        return this.nonProxyHosts;
+    }
+
+    /**
+     * Attempts to load a proxy from the environment.
+     *
+     * <p>
+     * Environment configurations are loaded in this oreder:
+     * <ol>
+     *     <li>Azure HTTPS</li>
+     *     <li>Azure HTTP</li>
+     *     <li>Java HTTPS</li>
+     *     <li>Java HTTP</li>
+     * </ol>
+     *
+     * Azure proxy configurations will be preferred over Java proxy configurations as they are more closely scoped to
+     * the purpose of the SDK. Additionally, more secure protocols, HTTPS vs HTTP, will be preferred.
+     *
+     * <p>
+     * {@code null} will be returned if no proxy was found in the environment.
+     *
+     * @param configuration The {@link Configuration} that is used to load proxy configurations from the environment.
+     * If {@code null} is passed then {@link Configuration#getGlobalConfiguration()} will be used. If
+     * {@link Configuration#NONE} is passed {@link IllegalArgumentException} will be thrown.
+     * @return A {@link ProxyOptions} reflecting a proxy loaded from the environment, if no proxy is found {@code null}
+     * will be returned.
+     * @throws IllegalArgumentException If {@code configuration} is {@link Configuration#NONE}.
+     */
+    public static ProxyOptions loadFromEnvironment(Configuration configuration) {
+        if (configuration == Configuration.NONE) {
+            throw new IllegalArgumentException("'configuration' cannot be 'Configuration.NONE'.");
```

Comment:
This should be wrapped with logger.logAsError().

Reply:
This will require either adding a static logger or create an instance every time this method is called, instead should this scenario return `null`? Throwing here was an arbitrary decision since passing no configuration won't return anything reasonable.